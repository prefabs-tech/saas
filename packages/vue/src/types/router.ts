import type { RouteComponent } from "vue-router";

interface RouteOverride {
  component?: RouteComponent;
  meta?: {
    layout?: RouteComponent;
  };
  path?: string;
}

interface RouteOverrides {
  acceptInvitation: RouteOverride;
  changePassword: RouteOverride;
  google?: RouteOverride;
  home?: string;
  login: RouteOverride;
  signup: RouteOverride & { disabled?: boolean };
  signupFirstUser: RouteOverride & { disabled?: boolean };
  passwordReset: RouteOverride;
  passwordResetRequest: RouteOverride;
  passwordResetRequestAcknowledge: RouteOverride;
  profile?: RouteOverride;
  roles?: RouteOverride;
  verifyEmail?: RouteOverride;
  verifyEmailReminder?: RouteOverride;
}

export type { RouteOverride, RouteOverrides };
