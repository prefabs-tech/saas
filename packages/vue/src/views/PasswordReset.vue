<template>
  <Page :title="t('user.passwordReset.title')" class="auth password-reset">
    <slot name="instructions"></slot>

    <PasswordResetForm @submit="handleSubmit" />
  </Page>
</template>

<script lang="ts">
export default {
  name: "PasswordReset",
};
</script>

<script setup lang="ts">
import { useI18n } from "@dzangolab/vue3-i18n";
import { Page } from "@dzangolab/vue3-ui";
import { useRouter } from "vue-router";

import PasswordResetForm from "../components/PasswordResetForm.vue";
import { useTranslations } from "../index";
import useUserStore from "../store";

import type { PasswordResetPayload } from "../types";

const messages = useTranslations();

const { t } = useI18n({ messages });

const { resetPassword } = useUserStore();

const router = useRouter();

const handleSubmit = async (payload: PasswordResetPayload) => {
  await resetPassword(payload).then(() => {
    router.push({ name: "login" });
  });
};
</script>
