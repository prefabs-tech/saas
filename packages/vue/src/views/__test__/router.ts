import { createRouter, createWebHistory } from "vue-router";

import Accounts from "../accounts/Index.vue";

const Test = {
  name: "Test",
  template: `<h1>Test</h1>`,
};

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    {
      component: Accounts,
      name: "accounts",
      path: "/accounts",
    },
    {
      component: Test,
      path: "/",
    },
  ],
});

export default router;
