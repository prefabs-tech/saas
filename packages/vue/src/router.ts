import { storeToRefs } from "pinia";
import { Router } from "vue-router";

import AuthGoogleCallback from "./components/AuthGoogleCallback.vue";
import useUserStore from "./store";
import { getVerificationStatus } from "./supertokens";
import AcceptInvitation from "./views/AcceptInvitation.vue";
import ChangePassword from "./views/ChangePassword.vue";
import VerifyEmailReminder from "./views/EmailVerificationReminder.vue";
import Login from "./views/Login.vue";
import PasswordReset from "./views/PasswordReset.vue";
import PasswordResetRequest from "./views/PasswordResetRequest.vue";
import PasswordResetRequestAcknowledge from "./views/PasswordResetRequestAcknowledge.vue";
import Profile from "./views/Profile.vue";
import Roles from "./views/Roles/Index.vue";
import Signup from "./views/Signup.vue";
import SignupFirstUser from "./views/SignupFirstUser.vue";
import VerifyEmail from "./views/VerifyEmail.vue";

import type {
  DzangolabVueUserConfig,
  RouteOverride,
  RouteOverrides,
} from "./types";
import type { RouteMeta, RouteRecordRaw } from "vue-router";

const _routes = {
  acceptInvitation: {
    component: AcceptInvitation,
    name: "acceptInvitation",
    path: "/signup/token/:token?",
  } as RouteRecordRaw,
  changePassword: {
    meta: {
      authenticated: true,
    } as RouteMeta,
    component: ChangePassword,
    name: "changePassword",
    path: "/change-password",
  },
  google: {
    component: AuthGoogleCallback,
    name: "authcallback",
    path: "/auth/callback/google",
  },
  login: {
    component: Login,
    name: "login",
    path: "/login",
  } as RouteRecordRaw,
  profile: {
    meta: {
      authenticated: true,
    } as RouteMeta,
    component: Profile,
    name: "profile",
    path: "/profile",
  } as RouteRecordRaw,
  roles: {
    meta: {
      authenticated: true,
    } as RouteMeta,
    component: Roles,
    name: "roles",
    path: "/roles",
  } as RouteRecordRaw,
  signup: {
    component: Signup,
    name: "signup",
    path: "/signup",
  } as RouteRecordRaw,
  signupFirstUser: {
    component: SignupFirstUser,
    name: "signupFirstUser",
    path: "/signup-first-user",
  } as RouteRecordRaw,
  passwordReset: {
    component: PasswordReset,
    name: "resetPassword",
    path: "/reset-password/:token?",
  } as RouteRecordRaw,
  passwordResetRequest: {
    component: PasswordResetRequest,
    name: "resetPasswordRequest",
    path: "/reset-password-request",
  } as RouteRecordRaw,
  passwordResetRequestAcknowledge: {
    component: PasswordResetRequestAcknowledge,
    name: "resetPasswordRequestAcknowledge",
    path: "/reset-password-request-acknowledge",
  },
  verifyEmail: {
    meta: {
      authenticated: true,
    } as RouteMeta,
    component: VerifyEmail,
    name: "verifyEmail",
    path: "/verify-email",
  },
  verifyEmailReminder: {
    meta: {
      authenticated: true,
    } as RouteMeta,
    component: VerifyEmailReminder,
    name: "verifyEmailReminder",
    path: "/verify-email-reminder",
  },
};

const getRoute = (
  defaultRoute: object,
  override?: RouteOverride,
): RouteRecordRaw => {
  return {
    ...defaultRoute,
    ...override,
  } as RouteRecordRaw;
};

const addRoutes = (router: Router, userConfig?: DzangolabVueUserConfig) => {
  const routes: RouteOverrides | undefined = userConfig?.routes;

  router.addRoute(getRoute(_routes.google, routes?.google));

  router.addRoute(getRoute(_routes.login, routes?.login));

  if (!routes?.signup?.disabled) {
    router.addRoute(getRoute(_routes.signup, routes?.signup));
  }

  if (routes?.signup?.disabled && !routes?.signupFirstUser?.disabled) {
    router.addRoute(getRoute(_routes.signupFirstUser, routes?.signupFirstUser));
  }

  router.addRoute(getRoute(_routes.passwordReset, routes?.passwordReset));

  router.addRoute(
    getRoute(_routes.passwordResetRequest, routes?.passwordResetRequest),
  );

  router.addRoute(getRoute(_routes.profile, routes?.profile));

  router.addRoute(getRoute(_routes.roles, routes?.roles));

  router.addRoute(
    getRoute(
      _routes.passwordResetRequestAcknowledge,
      routes?.passwordResetRequestAcknowledge,
    ),
  );

  router.addRoute(
    getRoute(
      _routes.changePassword,
      routes?.changePassword,
    ),
  );

  router.addRoute(getRoute(_routes.acceptInvitation, routes?.acceptInvitation));

  if (userConfig?.features?.signUp?.emailVerification) {
    router.addRoute(getRoute(_routes.verifyEmail, routes?.verifyEmail));

    router.addRoute(
      getRoute(_routes.verifyEmailReminder, routes?.verifyEmailReminder),
    );
  }
};

const redirectRoutes = (router: Router) => {
  router.beforeEach((to, from, next) => {
    const userStore = useUserStore();
    const { user } = storeToRefs(userStore);

    const routesToRedirect = [
      "acceptInvitation",
      "login",
      "signup",
      "signupFirstUser",
      "resetPassword",
      "resetPasswordRequest",
    ];
    const name = to.name as string;

    if (user.value && routesToRedirect.includes(name)) {
      next({ name: "profile" });
    } else {
      next();
    }
  });
};

const addAuthenticationGuard = (
  router: Router,
  userConfig?: DzangolabVueUserConfig,
) => {
  router.beforeEach(async (to) => {
    const meta = to.meta as RouteMeta;

    const userStore = useUserStore();

    const EmailVerificationEnabled =
      !!userConfig?.features?.signUp?.emailVerification;

    const name = to.name as string;
    const routesToRedirect = ["verifyEmail", "verifyEmailReminder"];
    const { user } = storeToRefs(userStore);
    const { getUser } = userStore;

    if (!user.value) {
      user.value = await getUser();
    }

    if (meta.authenticated && !user.value) {
      router.push({ name: "login" }); // using next inside async function is not allowed
    }

    if (EmailVerificationEnabled && user.value) {
      const isEmailVerified = await getVerificationStatus();

      if (
        meta.authenticated &&
        !isEmailVerified &&
        !routesToRedirect.includes(name)
      ) {
        router.push({ name: "verifyEmailReminder" });
      } else if (isEmailVerified && routesToRedirect.includes(name)) {
        router.push({ name: "home" });
      }
    }

    const isProfileCompleted = !!user.value?.isProfileCompleted;
    const profileCompletionEnabled = user.value?.isProfileCompleted !== undefined;

    if (meta.authenticated && profileCompletionEnabled && !isProfileCompleted && name !== "profile") {
      router.push({ name: "profile" });
    }
  });
};

const updateRouter = (router: Router, userConfig?: DzangolabVueUserConfig) => {
  addRoutes(router, userConfig);

  addAuthenticationGuard(router, userConfig);

  redirectRoutes(router);
};

export default updateRouter;
