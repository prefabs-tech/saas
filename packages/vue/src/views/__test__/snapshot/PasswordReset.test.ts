import configPlugin from "@dzangolab/vue3-config";
import i18Plugin, { useLocaleStore } from "@dzangolab/vue3-i18n";
import { LoadingButton, LoadingIcon } from "@dzangolab/vue3-ui";
import { mount, RouterLinkStub } from "@vue/test-utils";
import { createPinia } from "pinia";
import { describe, it, expect } from "vitest";

import appConfig from "../../../components/__test__/config";
import PasswordReset from "../../PasswordReset.vue";
import router from "../router";

import type { VueWrapper } from "@vue/test-utils";

describe("PasswordReset", () => {
  const pinia = createPinia();

  const { setLocale } = useLocaleStore(appConfig.slug);

  const locales = ["en", "fr"];

  for (const locale of locales) {
    setLocale(locale);

    const wrapper: VueWrapper = mount(PasswordReset, {
      global: {
        components: {
          LoadingButton,
          LoadingIcon,
        },
        plugins: [
          [
            configPlugin,
            {
              config: appConfig,
            },
          ],
          pinia,
          [
            i18Plugin,
            {
              config: appConfig,
            },
          ],
          router,
        ],
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
    });

    it("matches snapshot in " + locale, () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  }
});
