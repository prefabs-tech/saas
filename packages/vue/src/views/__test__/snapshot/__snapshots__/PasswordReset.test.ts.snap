// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PasswordReset > matches snapshot in en 1`] = `
<div
  class="page auth password-reset"
  data-centered="false"
>
  <div
    class="page-header"
  >
    <div
      class="page-title-wrapper"
    >
      <div
        class="title"
      >
        <h1>
          Enter your new password.
        </h1>
        
        <!---->
        
      </div>
      
      <!---->
      
    </div>
    <div
      class="page-toolbar"
    >
      
      
    </div>
  </div>
  <div
    class="page-content"
  >
    
    
    
    <form
      novalidate=""
    >
      <div
        class="field password"
      >
        <label
          for="password"
        >
          Enter password
        </label>
        
        <div
          class="password-input"
        >
          <input
            class=""
            name="password"
            placeholder=""
            tabindex="0"
            type="password"
          />
          <span
            class="eye-icon"
          >
            
            <img
              alt="eye-icon"
              src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3csvg%20fill='%23000000'%20width='800px'%20height='800px'%20viewBox='0%200%2032%2032'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M18.822%2022.904c-5.215%201.275-10.524-1.051-15.818-6.904%201.337-1.501%202.79-2.843%204.364-4.034l0.076-0.055c0.186-0.138%200.305-0.357%200.305-0.604%200-0.414-0.336-0.75-0.75-0.75-0.166%200-0.32%200.054-0.444%200.146l0.002-0.001c-1.91%201.447-3.588%203.024-5.086%204.761l-0.036%200.042c-0.115%200.131-0.185%200.305-0.185%200.494s0.070%200.363%200.186%200.495l-0.001-0.001c4.803%205.488%209.693%208.254%2014.582%208.254%201.123-0.001%202.212-0.142%203.252-0.406l-0.092%200.020c0.332-0.082%200.573-0.377%200.573-0.729%200-0.414-0.336-0.75-0.75-0.75-0.064%200-0.125%200.008-0.184%200.023l0.005-0.001zM16.75%2020c-0-0.414-0.336-0.75-0.75-0.75v0c-1.794-0.002-3.248-1.456-3.25-3.25v-0c0-0.414-0.336-0.75-0.75-0.75s-0.75%200.336-0.75%200.75v0c0.003%202.622%202.128%204.747%204.75%204.75h0c0.414-0%200.75-0.336%200.75-0.75v0zM23.565%2022.503c2.701-1.672%205.010-3.665%206.965-5.967l0.034-0.042c0.116-0.131%200.186-0.304%200.186-0.494s-0.070-0.363-0.187-0.495l0.001%200.001c-6.844-7.82-13.822-10.081-20.758-6.76l-7.277-7.276c-0.135-0.131-0.32-0.212-0.523-0.212-0.414%200-0.75%200.336-0.75%200.75%200%200.203%200.081%200.388%200.213%200.523l27.999%2028.001c0.136%200.136%200.324%200.22%200.531%200.22%200.415%200%200.751-0.336%200.751-0.751%200-0.207-0.084-0.395-0.22-0.531v0zM28.996%2016c-1.852%202.121-4.004%203.919-6.402%205.345l-0.121%200.067-2.636-2.635c0.569-0.767%200.911-1.731%200.912-2.776v-0c-0.003-2.622-2.128-4.747-4.75-4.75h-0c-1.045%200.002-2.009%200.344-2.789%200.921l0.013-0.009-2.29-2.29c6.027-2.647%2011.95-0.64%2018.062%206.127zM14.301%2013.239c0.486-0.307%201.077-0.489%201.711-0.489%201.788%200%203.238%201.45%203.238%203.238%200%200.634-0.182%201.225-0.497%201.724l0.008-0.013z'%3e%3c/path%3e%3c/svg%3e"
            />
            
          </span>
        </div>
        <!---->
        
      </div>
      <div
        class="field password"
      >
        <label
          for="password"
        >
          Re-enter password
        </label>
        
        <div
          class="password-input"
        >
          <input
            class=""
            name="confirmation"
            placeholder=""
            tabindex="0"
            type="password"
          />
          <span
            class="eye-icon"
          >
            
            <img
              alt="eye-icon"
              src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3csvg%20fill='%23000000'%20width='800px'%20height='800px'%20viewBox='0%200%2032%2032'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M18.822%2022.904c-5.215%201.275-10.524-1.051-15.818-6.904%201.337-1.501%202.79-2.843%204.364-4.034l0.076-0.055c0.186-0.138%200.305-0.357%200.305-0.604%200-0.414-0.336-0.75-0.75-0.75-0.166%200-0.32%200.054-0.444%200.146l0.002-0.001c-1.91%201.447-3.588%203.024-5.086%204.761l-0.036%200.042c-0.115%200.131-0.185%200.305-0.185%200.494s0.070%200.363%200.186%200.495l-0.001-0.001c4.803%205.488%209.693%208.254%2014.582%208.254%201.123-0.001%202.212-0.142%203.252-0.406l-0.092%200.020c0.332-0.082%200.573-0.377%200.573-0.729%200-0.414-0.336-0.75-0.75-0.75-0.064%200-0.125%200.008-0.184%200.023l0.005-0.001zM16.75%2020c-0-0.414-0.336-0.75-0.75-0.75v0c-1.794-0.002-3.248-1.456-3.25-3.25v-0c0-0.414-0.336-0.75-0.75-0.75s-0.75%200.336-0.75%200.75v0c0.003%202.622%202.128%204.747%204.75%204.75h0c0.414-0%200.75-0.336%200.75-0.75v0zM23.565%2022.503c2.701-1.672%205.010-3.665%206.965-5.967l0.034-0.042c0.116-0.131%200.186-0.304%200.186-0.494s-0.070-0.363-0.187-0.495l0.001%200.001c-6.844-7.82-13.822-10.081-20.758-6.76l-7.277-7.276c-0.135-0.131-0.32-0.212-0.523-0.212-0.414%200-0.75%200.336-0.75%200.75%200%200.203%200.081%200.388%200.213%200.523l27.999%2028.001c0.136%200.136%200.324%200.22%200.531%200.22%200.415%200%200.751-0.336%200.751-0.751%200-0.207-0.084-0.395-0.22-0.531v0zM28.996%2016c-1.852%202.121-4.004%203.919-6.402%205.345l-0.121%200.067-2.636-2.635c0.569-0.767%200.911-1.731%200.912-2.776v-0c-0.003-2.622-2.128-4.747-4.75-4.75h-0c-1.045%200.002-2.009%200.344-2.789%200.921l0.013-0.009-2.29-2.29c6.027-2.647%2011.95-0.64%2018.062%206.127zM14.301%2013.239c0.486-0.307%201.077-0.489%201.711-0.489%201.788%200%203.238%201.45%203.238%203.238%200%200.634-0.182%201.225-0.497%201.724l0.008-0.013z'%3e%3c/path%3e%3c/svg%3e"
            />
            
          </span>
        </div>
        <!---->
        
      </div>
      <div
        class="submit"
      >
        <button
          class="loading-button"
          data-v-3fdbc7b3=""
        >
          <div
            class="label"
            data-v-3fdbc7b3=""
          >
            <span
              data-v-3fdbc7b3=""
            >
              Change Password
            </span>
          </div>
          <!---->
        </button>
      </div>
    </form>
    
    <!---->
  </div>
</div>
`;

exports[`PasswordReset > matches snapshot in fr 1`] = `
<div
  class="page auth password-reset"
  data-centered="false"
>
  <div
    class="page-header"
  >
    <div
      class="page-title-wrapper"
    >
      <div
        class="title"
      >
        <h1>
          RÃ©initialisation de votre mot de passe
        </h1>
        
        <!---->
        
      </div>
      
      <!---->
      
    </div>
    <div
      class="page-toolbar"
    >
      
      
    </div>
  </div>
  <div
    class="page-content"
  >
    
    
    
    <form
      novalidate=""
    >
      <div
        class="field password"
      >
        <label
          for="password"
        >
          Nouveau mot de passe
        </label>
        
        <div
          class="password-input"
        >
          <input
            class=""
            name="password"
            placeholder=""
            tabindex="0"
            type="password"
          />
          <span
            class="eye-icon"
          >
            
            <img
              alt="eye-icon"
              src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3csvg%20fill='%23000000'%20width='800px'%20height='800px'%20viewBox='0%200%2032%2032'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M18.822%2022.904c-5.215%201.275-10.524-1.051-15.818-6.904%201.337-1.501%202.79-2.843%204.364-4.034l0.076-0.055c0.186-0.138%200.305-0.357%200.305-0.604%200-0.414-0.336-0.75-0.75-0.75-0.166%200-0.32%200.054-0.444%200.146l0.002-0.001c-1.91%201.447-3.588%203.024-5.086%204.761l-0.036%200.042c-0.115%200.131-0.185%200.305-0.185%200.494s0.070%200.363%200.186%200.495l-0.001-0.001c4.803%205.488%209.693%208.254%2014.582%208.254%201.123-0.001%202.212-0.142%203.252-0.406l-0.092%200.020c0.332-0.082%200.573-0.377%200.573-0.729%200-0.414-0.336-0.75-0.75-0.75-0.064%200-0.125%200.008-0.184%200.023l0.005-0.001zM16.75%2020c-0-0.414-0.336-0.75-0.75-0.75v0c-1.794-0.002-3.248-1.456-3.25-3.25v-0c0-0.414-0.336-0.75-0.75-0.75s-0.75%200.336-0.75%200.75v0c0.003%202.622%202.128%204.747%204.75%204.75h0c0.414-0%200.75-0.336%200.75-0.75v0zM23.565%2022.503c2.701-1.672%205.010-3.665%206.965-5.967l0.034-0.042c0.116-0.131%200.186-0.304%200.186-0.494s-0.070-0.363-0.187-0.495l0.001%200.001c-6.844-7.82-13.822-10.081-20.758-6.76l-7.277-7.276c-0.135-0.131-0.32-0.212-0.523-0.212-0.414%200-0.75%200.336-0.75%200.75%200%200.203%200.081%200.388%200.213%200.523l27.999%2028.001c0.136%200.136%200.324%200.22%200.531%200.22%200.415%200%200.751-0.336%200.751-0.751%200-0.207-0.084-0.395-0.22-0.531v0zM28.996%2016c-1.852%202.121-4.004%203.919-6.402%205.345l-0.121%200.067-2.636-2.635c0.569-0.767%200.911-1.731%200.912-2.776v-0c-0.003-2.622-2.128-4.747-4.75-4.75h-0c-1.045%200.002-2.009%200.344-2.789%200.921l0.013-0.009-2.29-2.29c6.027-2.647%2011.95-0.64%2018.062%206.127zM14.301%2013.239c0.486-0.307%201.077-0.489%201.711-0.489%201.788%200%203.238%201.45%203.238%203.238%200%200.634-0.182%201.225-0.497%201.724l0.008-0.013z'%3e%3c/path%3e%3c/svg%3e"
            />
            
          </span>
        </div>
        <!---->
        
      </div>
      <div
        class="field password"
      >
        <label
          for="password"
        >
          Confirmation de votre nouveau mot de passe
        </label>
        
        <div
          class="password-input"
        >
          <input
            class=""
            name="confirmation"
            placeholder=""
            tabindex="0"
            type="password"
          />
          <span
            class="eye-icon"
          >
            
            <img
              alt="eye-icon"
              src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3csvg%20fill='%23000000'%20width='800px'%20height='800px'%20viewBox='0%200%2032%2032'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M18.822%2022.904c-5.215%201.275-10.524-1.051-15.818-6.904%201.337-1.501%202.79-2.843%204.364-4.034l0.076-0.055c0.186-0.138%200.305-0.357%200.305-0.604%200-0.414-0.336-0.75-0.75-0.75-0.166%200-0.32%200.054-0.444%200.146l0.002-0.001c-1.91%201.447-3.588%203.024-5.086%204.761l-0.036%200.042c-0.115%200.131-0.185%200.305-0.185%200.494s0.070%200.363%200.186%200.495l-0.001-0.001c4.803%205.488%209.693%208.254%2014.582%208.254%201.123-0.001%202.212-0.142%203.252-0.406l-0.092%200.020c0.332-0.082%200.573-0.377%200.573-0.729%200-0.414-0.336-0.75-0.75-0.75-0.064%200-0.125%200.008-0.184%200.023l0.005-0.001zM16.75%2020c-0-0.414-0.336-0.75-0.75-0.75v0c-1.794-0.002-3.248-1.456-3.25-3.25v-0c0-0.414-0.336-0.75-0.75-0.75s-0.75%200.336-0.75%200.75v0c0.003%202.622%202.128%204.747%204.75%204.75h0c0.414-0%200.75-0.336%200.75-0.75v0zM23.565%2022.503c2.701-1.672%205.010-3.665%206.965-5.967l0.034-0.042c0.116-0.131%200.186-0.304%200.186-0.494s-0.070-0.363-0.187-0.495l0.001%200.001c-6.844-7.82-13.822-10.081-20.758-6.76l-7.277-7.276c-0.135-0.131-0.32-0.212-0.523-0.212-0.414%200-0.75%200.336-0.75%200.75%200%200.203%200.081%200.388%200.213%200.523l27.999%2028.001c0.136%200.136%200.324%200.22%200.531%200.22%200.415%200%200.751-0.336%200.751-0.751%200-0.207-0.084-0.395-0.22-0.531v0zM28.996%2016c-1.852%202.121-4.004%203.919-6.402%205.345l-0.121%200.067-2.636-2.635c0.569-0.767%200.911-1.731%200.912-2.776v-0c-0.003-2.622-2.128-4.747-4.75-4.75h-0c-1.045%200.002-2.009%200.344-2.789%200.921l0.013-0.009-2.29-2.29c6.027-2.647%2011.95-0.64%2018.062%206.127zM14.301%2013.239c0.486-0.307%201.077-0.489%201.711-0.489%201.788%200%203.238%201.45%203.238%203.238%200%200.634-0.182%201.225-0.497%201.724l0.008-0.013z'%3e%3c/path%3e%3c/svg%3e"
            />
            
          </span>
        </div>
        <!---->
        
      </div>
      <div
        class="submit"
      >
        <button
          class="loading-button"
          data-v-3fdbc7b3=""
        >
          <div
            class="label"
            data-v-3fdbc7b3=""
          >
            <span
              data-v-3fdbc7b3=""
            >
              Modifier votre mot de passe
            </span>
          </div>
          <!---->
        </button>
      </div>
    </form>
    
    <!---->
  </div>
</div>
`;
